### ISSUE TITLE
[Test failure] TC_AUTHOR_110: Certificate available date not working

### ISSUE THREAD

**Issue by angonz** [openedx/wg-build-test-release#429]
### Release

Sumac

### Expected behavior

Certificates should be available on the selected date, after the course end date

### Actual behavior

Certificates are made available after the course end date.

### Steps to reproduce

Work on an instructor-paced course. Enable the certificates.auto_certificate_generation waffle switch. 
In Studio, Settings, Schedule and Details, set a course end date.
Set "Certificate display behavior" to "A date after the course end date".
Set "Certificate available date" to a date after the end date. Save. 
![image](https://github.com/user-attachments/assets/dd1b2930-61e6-4484-82e5-e03f15d16e3c)

Go to the LMS. Complete the tests to get a passing grade.
In the progress page you should see a message "Certificate status. Final grades and any earned certificates are scheduled to be available after ..." 
The date informed should be the date set under "Certificate available date".
![image](https://github.com/user-attachments/assets/88f63bd1-b22b-4108-b408-2a9486b7d57b)

**Comment by AntonellaPerrupato** [comment]
@mariajgrimaldi  i was able to check this

**Comment by mariajgrimaldi** [comment]
I used the student account in the sandbox and this course: https://apps.sumac.demo.edly.io/learning/course/course-v1:Test+test_cert+2024/dates, which I'm guessing is the same one used during this test.

I see the correct date in the dates section:
![image](https://github.com/user-attachments/assets/1544e173-6957-4958-ab0b-7d024e77d37a)

But the course status is less accurate, saying the certificates will be available after the course ends but with no specific date:

![image](https://github.com/user-attachments/assets/73c009c6-9e29-4fbe-bb98-7e53e36ad81c)

Sometimes changes take a while to be reflected in the LMS, so _maybe_ that's what happened during the test. I also tested with a new student, and I can see that the dates I mentioned are displayed. I played around a bit more and generated an honor certificate through the Django admin (the enrollments of these users are in audit mode, so [no certificate would be generated](https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/enable_certificates.html#discontinue-audit-track-certificates) as far as I understand), then here's what I see in the certificate status section:

![image](https://github.com/user-attachments/assets/5be738fa-172a-454c-a9fd-27db87d22e3f)

**Comment by crathbun428** [comment]
If I'm reading this correctly, there is never **incorrect** information being communicated. Either we say after <course end date> or after exact date. Both of which are accurate. I agree this is a bug that should be addressed, but I don't think this rises to the level of a release blocker. @mariajgrimaldi, @angonz, @AntonellaPerrupato - do any of you disagree? I can be swayed, but this would be my read of the situation.

**Comment by angonz** [comment]
The reason is that the certificate data in the courseware api is None if the user didn't earn a certificate. This is processed by the [get_cert_data](https://github.com/openedx/edx-platform/blob/0bd0e6f4cab4d398216f010b6cd09ab21e7297e2/lms/djangoapps/courseware/views/views.py#L1106) function in edx-platform. If the user didn't get a certificate yet, then the Progress page will display the [course end date](https://github.com/openedx/frontend-app-learning/blob/b09bcbd3aef5f4bc81c498ccd56347b83b26c4fa/src/courseware/course/course-exit/CourseCelebration.jsx#L265) instead, as the course metadata does not contain the certificate available date. When the user has a certificate, although it's not available yet, there is a certificate data in the course metadata which includes the [certificate available date](https://github.com/openedx/frontend-app-learning/blob/b09bcbd3aef5f4bc81c498ccd56347b83b26c4fa/src/courseware/course/course-exit/CourseCelebration.jsx#L133-L134).

I agree that the information displayed is not incorrect. Although it would be disappointing for a student to take a course expecting my certificate by the course end date, and then after earning it, being informed that it will be available days after. Moreover, the actual certificate available date is known and does not depend on the student, so I don't see the point of hiding it.

Overall, I think that the whole certificate workflow should be revised as it is very cumbersome...

**Comment by mariajgrimaldi** [comment]
Thanks, @crathbun428! I agree this should be addressed somehow. Can we clarify the correct behavior in this case?

Thank you, @angonz, for the thorough explanation! That'd give us a sense of where to look once we start addressing this behavior.

**Comment by crathbun428** [comment]
@mariajgrimaldi - wait - am I reading it right that the correct dates appear when you log in as a learner? If so, I would say nothing needs to be fixed here?

What am I missing? @angonz

**Comment by angonz** [comment]
No, the information is correct only in the Dates page. In the Progress page it's the same. No matter if you log in as student or staff.

**Comment by mariajgrimaldi** [comment]
What I understand is that although nothing needs to be fixed, it's not a great experience for the student. Is my interpretation correct, @angonz? Should we escalate this to the product WG if that's the case?

**Comment by crathbun428** [comment]
@angonz and @mariajgrimaldi - 2U put a product proposal in to fix this, [here](https://github.com/orgs/openedx/projects/4/views/20?pane=issue&itemId=96342800&issue=openedx%7Cplatform-roadmap%7C423). This is the same issue, correct? (It looked the same to me on first glance). How can I best link these two issues to avoid duplication of efforts?

**Comment by angonz** [comment]
Hi @crathbun428,
I don't think so. As far as I understood, 2U's proposal is to change the certificate **issued date** as the course end date instead of the certificate modified date, for instructor paced courses where the certificate display behavior is set as 'Course end date'. 

This issue is for courses where the certificate display behavior is set as 'A date after the course end date' and is about the  **certificate available date** displayed in the progress page.

**Comment by crathbun428** [comment]
Got it. That makes sense.

Then this issue needs to be fixed separately. Desired behavior for instructor-based courses where the Course Delivery Team has:  

1. Set "Certificate display behavior" to: "A date after the course end date" and  
2. Set "Certificate available date" to a date after the end date
(All of the above is done in Studio, Settings, Schedule and Details)

The Progress page for the course correctly reflects the Certificate available date in the Certificate Status section of the page. It should read: This course ends on <insert Course end date>. Final grades and any earned certificates are scheduled to be available after <insert Certificate available date>.

Sound good? I believe this captures the entire fix needed here? @mariajgrimaldi @angonz

**Comment by dcoa** [comment]
Hello @angonz @mariajgrimaldi , I opened 2 PRs (one in the backend, and one in the frontend) to solve this. I added you as a  reviewers, expecting your insightful feedback
